src/openforms/emails/tests/test_digest_functions.py

```diff
@@ -487,3 +530,215 @@ def test_both_expiring_and_invalid_certificates_are_collected(self):
+            form=form,
+            name="Variable 1",
+            key="foo",
+            user_defined=True,
+        )
+        FormLogicFactory(
+            form=form,
+            json_logic_trigger={"==": [{"var": "foo"}, "apple"]},
+        )
+
+        invalid_logic_rules = collect_invalid_logic_rules()
+
+        logs_exist = TimelineLogProxy.objects.exists()
+
+        self.assertEqual(len(invalid_logic_rules), 0)
+        self.assertFalse(logs_exist)
+
+    def test_invalid_logic_rules_with_nested_key_are_only_logged(self):
+        form = FormFactory.create()
+        FormVariableFactory.create(
+            form=form,
+            name="Variable 1",
+            key="foo",
+            user_defined=True,
+            data_type=FormVariableDataTypes.object,
+        )
+        FormLogicFactory(
+            form=form,
+            json_logic_trigger={
+                "and": [
+                    {"<": [{"var": "foo.bar"}, "apple"]},
+                    {"<": [{"var": "foo.0.bar"}, "apple"]},
+                ]
+            },
+            actions=[
+                {
+                    "action": {"type": "variable", "value": "d"},
+                    "variable": "foo.wrong",
+                    "component": "",
+                    "form_step": "",
+                    "form_step_uuid": "c490b80f-05fe-4006-884d-f631c8abe968",
+                },
+            ],
+        )
+
+        with self.assertLogs() as logs:
+            collect_invalid_logic_rules()
+
+        logs_messages = [log.getMessage() for log in logs.records]
+
+        self.assertEqual(len(logs.records), 3)
+        self.assertIn(
+            f"invalid variable reference (foo.wrong) in logic of form {form.admin_name}",
+            logs_messages,
+        )
+        self.assertIn(
+            f"invalid variable reference (foo.bar) in logic of form {form.admin_name}",
+            logs_messages,
+        )
+        self.assertIn(
+            f"invalid variable reference (foo.0.bar) in logic of form {form.admin_name}",
+            logs_messages,
+        )
+
+    def test_invalid_logic_rules_are_collected_and_not_logged(self):
+        form = FormFactory.create()
+        FormLogicFactory(
+            form=form,
+            json_logic_trigger={
+                "and": [
+                    {"<": [{"var": "temp"}, 110]},
+                    {"<": [{"var": "temp.0.other"}, 110]},
+                    {"<": [{"var": {"var": "another"}}, 110]},
+                    {"<": [{"var": ""}, 110]},
+                    {"==": [{"var": "pie.filling"}, "apple"]},
+                ]
+            },
+            actions=[
+                {
+                    "action": {"type": "variable", "value": "d"},
+                    "variable": "foo.wrong",
+                    "component": "",
+                    "form_step": "",
+                    "form_step_uuid": "c490b80f-05fe-4006-884d-f631c8abe968",
+                },
+            ],
+        )
```

Comment:
I'm missing some cases:

* variable `foo` of type string & logic rule doing `foo.bar`
* variable `foo` of type array & logic rule doing `foo.bar`